test/functional/api/v2/job_invocations_controller_test.rb

```diff
@@ -23,45 +23,60 @@ class JobInvocationsControllerTest < ActionController::TestCase
-        assert_response :success
-      end
+        test 'should create valid with job_template_id' do
+          post :create, params: { job_invocation: @attrs }
 
-      test 'should create with description format overridden' do
-        attrs = { :job_category => @template.job_category, :name => 'RandomName', :job_template_id => @template.id,
-                  :targeting_type => 'static_query', :search_query => 'foobar', :description_format => 'format' }
-        post :create, params: { :job_invocation => attrs }
+          invocation = ActiveSupport::JSON.decode(@response.body)
+          assert_equal @attrs[:job_category], invocation['job_category']
+          assert_response :success
+        end
 
-        invocation = ActiveSupport::JSON.decode(@response.body)
-        assert_equal attrs[:description_format], invocation['description']
-      end
+        test 'should create with description format overridden' do
+          @attrs[:description_format] = 'format'
+          post :create, params: { job_invocation: @attrs }
 
-      test 'should create with recurrence' do
-        attrs = { :job_category => @template.job_category, :name => 'RandomName',
-                  :job_template_id => @template.id,:targeting_type => 'static_query',
-                  :search_query => 'foobar', :recurrence => {:cron_line => '5 * * * *'}}
+          invocation = ActiveSupport::JSON.decode(@response.body)
+          assert_equal @attrs[:description_format], invocation['description']
+        end
 
-        post :create, params: { :job_invocation => attrs }
-        invocation = ActiveSupport::JSON.decode(@response.body)
-        assert_equal invocation['mode'], 'recurring'
-        assert_response :success
-      end
+        test 'should create with recurrence' do
+          @attrs[:recurrence] = { cron_line: '5 * * * *' }
+          post :create, params: { job_invocation: @attrs }
+          invocation = ActiveSupport::JSON.decode(@response.body)
+          assert_equal invocation['mode'], 'recurring'
+          assert_response :success
+        end
 
-      test 'should create with schedule' do
-        attrs = { :job_category => @template.job_category, :name => 'RandomName',
-                  :job_template_id => @template.id,:targeting_type => 'static_query',
-                  :search_query => 'foobar', :scheduling => {:start_at => Time.now.to_s}}
+        test 'should create with schedule' do
+          @attrs[:scheduling] = { start_at: Time.now.to_s }
+          post :create, params: { job_invocation: @attrs }
+          invocation = ActiveSupport::JSON.decode(@response.body)
+          assert_equal invocation['mode'], 'future'
+          assert_response :success
+        end
 
-        post :create, params: { :job_invocation => attrs }
-        invocation = ActiveSupport::JSON.decode(@response.body)
-        assert_equal invocation['mode'], 'future'
-        assert_response :success
+        test 'should create with feature' do
```

Comment:
This is the only test I added, I just refactored the test to take advantage of a global `@attrs`